drivers/serial/uart_nrfx_uarte.c

```diff
@@ -2156,93 +2172,70 @@ static void wait_for_tx_stopped(const struct device *dev)
 	}
 }
 
-
-static int uarte_nrfx_pm_action(const struct device *dev,
-				enum pm_device_action action)
+static void uarte_pm_resume(const struct device *dev)
 {
-	NRF_UARTE_Type *uarte = get_uarte_instance(dev);
-#if defined(UARTE_ANY_ASYNC) || defined(UARTE_INTERRUPT_DRIVEN)
-	struct uarte_nrfx_data *data = dev->data;
-#endif
 	const struct uarte_nrfx_config *cfg = dev->config;
-	int ret;
 
-#ifdef UARTE_ANY_ASYNC
-	/* If low power mode for asynchronous mode is used then there is nothing to do here.
-	 * In low power mode UARTE is turned off whenever there is no activity.
-	 */
-	if (data->async && (cfg->flags & UARTE_CFG_FLAG_LOW_POWER)) {
-		return 0;
-	}
-#endif
+	(void)pinctrl_apply_state(cfg->pcfg, PINCTRL_STATE_DEFAULT);
 
-	switch (action) {
-	case PM_DEVICE_ACTION_RESUME:
-		ret = pinctrl_apply_state(cfg->pcfg, PINCTRL_STATE_DEFAULT);
-		if (ret < 0) {
-			return ret;
-		}
+	if (!LOW_POWER_ENABLED(cfg)) {
```

Comment:
For `CONFIG_PM_DEVICE_RUNTIME`, low power flags should be either ignored here or not set at all in `dev->config`.

Reply:
added check for runtime PM.